gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagTask.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.orchestration.task;
+
+import java.io.IOException;
+
+import org.apache.gobblin.annotation.Alpha;
+import org.apache.gobblin.runtime.api.MultiActiveLeaseArbiter;
+import org.apache.gobblin.service.modules.orchestration.DagManagementStateStore;
+import org.apache.gobblin.service.modules.orchestration.DagTaskVisitor;
+import org.apache.gobblin.service.modules.orchestration.processor.DagProc;
+
+
+/**
+ * Defines an individual task or job in a Dag.
+ * Upon completion of the {@link DagProc#process(DagManagementStateStore)} it will mark the lease
+ * acquired by {@link org.apache.gobblin.runtime.api.MultiActiveLeaseArbiter} as complete
+ * @param <T>
+ */
+
+@Alpha
+public abstract class DagTask<T> {
+
+  protected MultiActiveLeaseArbiter.LeaseObtainedStatus leaseObtainedStatusStatus;
+
+  /**
+   * Currently, conclusion of {@link DagTask} marks and records a successful release of lease.
+   * It is invoked after {@link DagProc#process(DagManagementStateStore)} is completed successfully.
+   * @param multiActiveLeaseArbiter
+   * @throws IOException
+   */
```

Comment:
This is different from our approach to launch in that we finish the action before completing lease. Can you document the pros and cons of this method? For ex pro we don't have to do lease arbitration twice for attempting the action and doing it. However now the lease validity time needs to be much larger and include the time to contact executor and carry out the action. Let's note these details here or somewhere else appropriate.

Reply:
I think there's some confusion here. The lease acquired for a `DagTask` happens after we have received or processed a message from the CDC stream based on what action needs to be taken on the Dag. Now, we acquire the lease... do our processing on the Dag and upon completion release the lease ... marking it as successful. I agree with the part that lease validity time needs to be increased

Reply:
Ah you're right I am getting this multi-active part confused with phase 2. Let's add some comments or javadoc to denote potential failures and/or steps needed to complete this event ie: contact executor over network and response time